### ISSUE TITLE
Ability to load .magnet files

### ISSUE THREAD

**Issue by peterbuga** [tympanix/Electorrent#145]
**Client:**
- [ ] µTorrent
- [ ] qBittorrent
- [x] Transmission
- [ ] rTorrent

**Description:**
This is not an issue but more like a quick improvement (i think). So far I'm able to drag'n'drop .torrent files with not problem and also load mangnet links from clipboard. 
In my current setup I have a torrent blackhole that saves .magnet links when a torrent is not found, I think a quick check if the dragged file has an .magnet extension + valid magnet format will improve a bit the app.

Thanks for all the hard work!

**Comment by tympanix** [comment]
That is a good idea. The more use cases we can cover the better. Do you know anything about the encoding and structure of a `.magnet` file? I assume it contains a single `utf-8` encoded line with the magnet URL itself?

**Comment by peterbuga** [comment]
I assume you're right. 
I have to admit I've done 0 reading to the code to see exactly what happens behind the scenes (i'm not fluent in latest in practices 😬) but for example if I open the .magnet file in a text editor, copy the the whole magnet link from there (which is exactly the one you will get from a normal website) and then CMD + I (macOS) it loads perfectly.

So I was thinking, the same way you load/read a `.torrent` file + the way you load a clipboard-magnet url you could add the ability to load `.magnet` files too (shouldn't be too hard, in theory at least). let me know if my explanation makes any sense 😅

**Comment by tympanix** [comment]
Yes that makes sense. Following your description the file seems to contain only the magnet link itself. I would be able to use existing functionality like you mention above. Thanks for reporting :+1:

**Comment by peterbuga** [comment]
thanks also for looking into it 👍!
